docs/addons.md

```diff
@@ -0,0 +1,288 @@
+
+The [linkerd documentation][linkerd-docs] suggests the following for installing the command line tool:
+
+```bash
+curl -sL https://run.linkerd.io/install | sh
+```
+
+You should also make sure the `KUBECONFIG` environment variable is set to point to the MicroK8s config file:
+
+```bash
+export KUBECONFIG=$KUBECONFIG:/snap/microk8s/current/client.config
+```
+
+You can verify that linkerd is operating by running:
+
+```bash
+linkerd check
+```
+
+For more on general usage for linkerd, see the [linkerd documentation].
+
+
+### metrics-server
+
+The metrics-server addon is a complete implementation of the Kubernetes metrics
+server project. Enabling the metrics-server will provide API access to
+continuously updated metrics for MicroK8s.
+
+More details on the scope and planned functionality of the metrics-server can be found in its [design documentation][metrics-design-doc].
+
+### prometheus
+
+Prometheus is a popular way to monitor a Kubernetes system, and one which is
+easy to integrate with MicroK8s. To use Grafana to view the collected data,
+it is advisable to also enable the `dashboard` addon at the same time:
+
+```bash
+microk8s.enable dashboard prometheus
+```
+
+You can verify the Grafana dashboard is available by running:
+
+```bash
+microk8s.kubectl cluster-info
+```
+
+Now, run the proxy service
+
+```
+microk8s.kubectl proxy &
+```
+(You may wish to run the proxy in a different shell rather than as a background process).
+
+You can now navigate to the Grafana dashboard at the address:
+<http://127.0.0.1:8001/api/v1/namespaces/kube-system/services/monitoring-grafana/proxy>
+
+
+![IMAGE Grafana](#ref)
+
+
+
+### rbac
+
+When enabled, the kube-apiserver `authorization-mode` will be set to `RBAC`. Kubernetes will then automatically add all the default roles and rolebindings.
+
+When disabled, the addon will retrun the `authorization-mode` to `AlwaysAllow`.
+
+For more information on using RBAC and the roles and bindings, see the
+[upstream Kubernetes documentation][kubernetes-rbac].
+
+### registry
+
+The registry shipped with MicroK8s is hosted within the Kubernetes cluster and
+is exposed as a NodePort service on port 32000 of the localhost. Note that this
+is an insecure registry and you may need to take extra steps to limit access to
+it.
+
+For more information on using this private registry, please see the
+[Working with registries documentation](#ref)
+
+
+### storage
+
+This storage class makes use of the hostpath-provisioner pointing to a
+directory on the host. Persistent volumes are created under
+`${SNAP_COMMON}/default-storage`. On an Ubuntu system this is commonly
+`/var/snap/microk8s/common/default-storage/`.
+
+If this addon is subsequently disabled, you will be asked if you wish to
+permanently remove any storage which may have been created.
+
+
```

Comment:
cilium and helm are the latest add-ons.  I do not have any documentation for these, only the reference created here https://github.com/canonical-web-and-design/microk8s.io/blob/master/docs/index.md